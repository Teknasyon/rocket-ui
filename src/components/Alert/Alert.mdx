import { Canvas, Meta, Story, Controls } from '@storybook/blocks';

import * as AlertStories from './Alert.stories';

<Meta of={AlertStories} />

# Alert

Alert is a component that displays a message to the user. It is often used to display information, warning, error, or success messages.

- [Overview](#overview)

- [Playground](#playground)

- [Usage with props](#usage)

- [Variants](#variants)

- [Customizable Slots](#customizable-slots)

- [Tips](#tips)

## Overview <a id="overview" />

An Alert component is a user interface (UI) element that is used to display important information to the user. It can be used to provide feedback, success or error messages, or to simply draw the user's attention to something. Alert components are typically used in situations where the user needs to be notified of something immediately, or where the information being displayed is important and requires the user's attention.

For example, an Alert component could be used to display a success message after the user has completed a form or submitted some data, to display an error message if there is a problem with the user's input, to display a warning if the user is about to perform an action that could have unintended consequences, or just to pass important information to the user.

In general, Alert components should be used sparingly, as overusing them can make your UI cluttered and difficult to read. It's important to use Alert components only when necessary, and to provide clear and concise messages that help the user understand what is happening.

<Canvas>
  <Story of={AlertStories.Overview} />
</Canvas>

## Playground <a id="playground" />

> Changes you make in the controls will be reflected in the example above. Try it yourself!

<Controls of={AlertStories.Overview} exclude={/^(click|close|on.*)$/} />

## Usage with props <a id="usage" />

### type (required)

The **type** prop determines the variant of the Alert component. The following values are accepted:

- **success**: Indicates that a successful action has occurred.
- **error**: Indicates that an error has occurred.
- **warning**: Indicates that the user should be cautious about something.
- **info**: Provides additional information to the user.

### title (optional)

The **title** prop specifies the title of the Alert component. It should be short and concise, and should clearly convey the purpose of the Alert.

## description (optional)

The **description** prop provides a longer description of the Alert component. It can be used to provide more detailed information or instructions to the user.

### size (optional)

The **size** prop determines the size of the Alert component. The following values are accepted:

- **small**: Renders a small Alert component.
- **medium**: Renders a medium-sized Alert component.
- **large**: Renders a large Alert component.

### closable (optional)

If **closable** prop is set to true, the Alert component will include a close button that the user can use to dismiss the Alert.

## Variants <a id="variants" />

> Our system has 4 Alert variants: **Success**, **Error**, **Information** and **Warning**.

### Success

<Canvas>
  <Story of={AlertStories.Success} />
</Canvas>

### Error

<Canvas>
  <Story of={AlertStories.Error} />
</Canvas>

### Information

<Canvas>
  <Story of={AlertStories.Information} />
</Canvas>

### Warning

<Canvas>
  <Story of={AlertStories.Warning} />
</Canvas>

## Customizable Slots <a id="customizable-slots" />

### `#close`

> This `close` slot can be used for creating your own custom close buttons like this one example below ðŸ‘‡

<Canvas>
  <Story of={AlertStories.CustomCloseButton} />
</Canvas>

### Tips <a id="tips" />

ðŸ’¡ Use the title prop to provide a clear and concise message to the user.

ðŸ’¡ Use the description prop to provide more detailed information or instructions, if necessary.

ðŸ’¡ Choose the appropriate variant (success, error, warning, or info) for the Alert based on the content it is displaying.

ðŸ’¡ Keep the Alert small and concise. If you need to provide a lot of information, consider using a separate component or modal instead of trying to cram everything into the Alert.

ðŸ’¡ If the Alert is closable, make sure to provide clear instructions on how to dismiss it (e.g. "Click the 'X' to close this Alert").
