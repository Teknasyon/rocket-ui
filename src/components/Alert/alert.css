@tailwind components;

.alert {
  @apply w-full font-sans relative p-4 rounded text-sm flex justify-between transition-transform duration-1000 box-border;
}

.alert--small {
  @apply p-2 text-xs;
}

.alert-success {
  @apply bg-green-100 text-green-500;
}

.alert-error {
  @apply bg-red-100 text-red-500;
}

.alert-warning {
  @apply bg-yellow-100 text-yellow-500;
}

.alert-info {
  @apply bg-blue-100 text-blue-500;
}

.icon {
  @apply grid place-items-start content-center;
}

.texts {
  @apply w-full grid place-items-start content-center;
}

.texts__title {
  @apply font-bold;
}

.texts__description {
  @apply text-gray-600;
}

.close {
  @apply grid place-items-end content-center;
}

.close__button {
  @apply text-gray-600;
}

.alert-success .close__button {
  @apply text-white bg-green-500 hover:bg-green-500;
}

.alert-info .close__button {
  @apply text-white bg-blue-500 hover:bg-blue-500;
}

.alert-warning .close__button {
  @apply text-white bg-yellow-500 hover:bg-yellow-500;
}

.alert-error .close__button {
  @apply text-white bg-red-500 hover:bg-red-500;
}
