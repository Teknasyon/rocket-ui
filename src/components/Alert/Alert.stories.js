import { action } from '@storybook/addon-actions';
import Alert from './RAlert.vue';
import Icon from '../Icon/RIcon.vue';

const Default = (args) => ({
  components: { Alert },
  setup() {
    return { args, click: action('close') };
  },
  template: `<Alert v-bind="args" @close="click"/>`,
});

const DefaultWithSlots = (args) => ({
  components: { Alert, Icon },
  setup() {
    return { args, onClick: action('close') };
  },
  template: `
  <Alert v-bind="args">
    <template #close>
      <Icon name="logout" @click="onClick" />
    </template>
  </Alert>`,
});

const defaultArgs = {
  type: {
    control: {
      type: 'select',
      options: ['success', 'info', 'warning', 'error'],
    },
    defaultValue: 'success',
    table: {
      defaultValue: {
        summary: 'success',
        detail: 'success | info | warning | error',
      },
    },
    description: 'Type of alert',
  },
  title: {
    control: {
      type: 'text',
    },
    defaultValue: 'Alert Title',
  },
  description: {
    control: {
      type: 'text',
    },
    defaultValue:
      'A type of alert which appears in a layer above other content, visually similar to a mobile or desktop push notification.',
  },
  size: {
    control: {
      type: 'select',
      options: ['small', 'medium', 'large'],
    },
    defaultValue: 'medium',
    table: {
      defaultValue: {
        summary: 'medium',
        detail: 'small | medium | large ',
      },
    },
    description: 'Sizes of alert',
  },
  closable: {
    control: {
      type: 'boolean',
    },
    defaultValue: true,
  },
  closeText: {
    control: {
      type: 'text',
    },
    defaultValue: 'Close',
  },
};

export default {
  title: 'Components/Alert',
  component: Alert,

  argTypes: {
    ...defaultArgs,
  },

  parameters: {
    viewMode: 'docs',
  },
};

export const Overview = {
  render: Default.bind({}),
  name: 'Overview',

  argTypes: {
    ...Default.argTypes,
  },

  args: {
    type: 'success',
  },
};

export const Success = {
  render: Default.bind({}),
  name: 'Success',

  argTypes: {
    ...Default.argTypes,
  },

  args: {
    type: 'success',
    title: 'Success',
    description:
      'Success Variant! This is a short but descriptive message that indicates a successful action.',
  },
};

export const Error = {
  render: Default.bind({}),
  name: 'Error',

  argTypes: {
    ...Default.argTypes,
  },

  args: {
    type: 'error',
    title: 'Error',
    description:
      'Error Variant! This is a short but descriptive message that indicates a negative action.',
  },
};

export const Information = {
  render: Default.bind({}),
  name: 'Information',

  argTypes: {
    ...Default.argTypes,
  },

  args: {
    type: 'info',
    title: 'Info',
    description:
      'Info Variant! This is a short but descriptive message that indicates a neutral informative action.',
  },
};

export const Warning = {
  render: Default.bind({}),
  name: 'Warning',

  argTypes: {
    ...Default.argTypes,
  },

  args: {
    type: 'warning',
    title: 'Warning',
    description:
      'Warning Variant! This is a short but descriptive message that indicates a negative or dangerous action.',
  },
};

export const CustomCloseButton = {
  render: DefaultWithSlots.bind({}),
  name: 'Custom Close Button',

  argTypes: {
    ...Default.argTypes,
  },

  args: {
    type: 'info',
    title: 'With a custom button',
    description:
      'This alert box looks very similar to our default examples but it contains a custom close button in it!',
  },
};
