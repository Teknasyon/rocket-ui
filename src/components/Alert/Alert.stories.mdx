import { Canvas, Meta, Story, ArgsTable } from '@storybook/addon-docs';
import { action } from '@storybook/addon-actions';
import Alert from './Alert.vue';
import Icon from '../Icon/Icon.vue';

export const Default = (args) => ({
  components: { Alert },
  setup() {
    return { args, click: action('close') };
  },
  template: `<Alert v-bind="args" @close="click"/>`,
});

export const DefaultWithSlots = (args) => ({
  components: { Alert, Icon },
  setup() {
    return { args, onClick: action('close') };
  },
  template: `
  <Alert v-bind="args">
    <template #close>
      <Icon name="logout" @click="onClick" />
    </template>
  </Alert>`,
});

export const defaultArgs = {
  type: {
    control: {
      type: 'select',
      options: ['success', 'info', 'warning', 'error'],
    },
    defaultValue: 'success',
    table: {
      defaultValue: {
        summary: 'success',
        detail: 'success | info | warning | error',
      },
    },
    description: 'Type of alert',
  },
  title: {
    control: {
      type: 'text',
    },
    defaultValue: 'Alert Title',
  },
  description: {
    control: {
      type: 'text',
    },
    defaultValue:
      'A type of alert which appears in a layer above other content, visually similar to a mobile or desktop push notification.',
  },
  size: {
    control: {
      type: 'select',
      options: ['small', 'medium', 'large'],
    },
    defaultValue: 'medium',
    table: {
      defaultValue: {
        summary: 'medium',
        detail: 'small | medium | large ',
      },
    },
    description: 'Sizes of alert',
  },
  closable: {
    control: {
      type: 'boolean',
    },
    defaultValue: true,
  },
  closeText: {
    control: {
      type: 'text',
    },
    defaultValue: 'Close',
  },
};

<Meta
  title="Components/Alert"
  component={Alert}
  argTypes={{ ...defaultArgs }}
  parameters={{
    viewMode: 'docs',
  }}
/>

# Alert

Alert is a component that displays a message to the user. It is often used to display information, warning, error, or success messages.

- [Overview](#overview)

- [Playground](#playground)

- [Usage with props](#usage)

- [Variants](#variants)

- [Customizable Slots](#customizable-slots)

- [Tips](#tips)

## Overview <a id="overview" />

An Alert component is a user interface (UI) element that is used to display important information to the user. It can be used to provide feedback, success or error messages, or to simply draw the user's attention to something. Alert components are typically used in situations where the user needs to be notified of something immediately, or where the information being displayed is important and requires the user's attention.

For example, an Alert component could be used to display a success message after the user has completed a form or submitted some data, to display an error message if there is a problem with the user's input, to display a warning if the user is about to perform an action that could have unintended consequences, or just to pass important information to the user.

In general, Alert components should be used sparingly, as overusing them can make your UI cluttered and difficult to read. It's important to use Alert components only when necessary, and to provide clear and concise messages that help the user understand what is happening.

<Canvas withSource="open">
  <Story
    name="Overview"
    argTypes={{ ...Default.argTypes }}
    args={{ type: 'success' }}
  >
    {Default.bind({})}
  </Story>
</Canvas>

## Playground <a id="playground" />

> Changes you make in the controls will be reflected in the example above. Try it yourself!

<ArgsTable story="Overview" exclude={/^(click|close|on.*)$/} />

## Usage with props <a id="usage" />

### type (required)

The **type** prop determines the variant of the Alert component. The following values are accepted:

- **success**: Indicates that a successful action has occurred.
- **error**: Indicates that an error has occurred.
- **warning**: Indicates that the user should be cautious about something.
- **info**: Provides additional information to the user.

### title (optional)

The **title** prop specifies the title of the Alert component. It should be short and concise, and should clearly convey the purpose of the Alert.

## description (optional)

The **description** prop provides a longer description of the Alert component. It can be used to provide more detailed information or instructions to the user.

### size (optional)

The **size** prop determines the size of the Alert component. The following values are accepted:

- **small**: Renders a small Alert component.
- **medium**: Renders a medium-sized Alert component.
- **large**: Renders a large Alert component.

### closable (optional)

If **closable** prop is set to true, the Alert component will include a close button that the user can use to dismiss the Alert.

## Variants <a id="variants" />

> Our system has 4 Alert variants: **Success**, **Error**, **Information** and **Warning**.

### Success

<Canvas withSource="open">
  <Story
    name="Success"
    argTypes={{ ...Default.argTypes }}
    args={{
      type: 'success',
      title: 'Success',
      description:
        'Success Variant! This is a short but descriptive message that indicates a successful action.',
    }}
  >
    {Default.bind({})}
  </Story>
</Canvas>

### Error

<Canvas withSource="open">
  <Story
    name="Error"
    argTypes={{ ...Default.argTypes }}
    args={{
      type: 'error',
      title: 'Error',
      description:
        'Error Variant! This is a short but descriptive message that indicates a negative action.',
    }}
  >
    {Default.bind({})}
  </Story>
</Canvas>

### Information

<Canvas withSource="open">
  <Story
    name="Information"
    argTypes={{ ...Default.argTypes }}
    args={{
      type: 'info',
      title: 'Info',
      description:
        'Info Variant! This is a short but descriptive message that indicates a neutral informative action.',
    }}
  >
    {Default.bind({})}
  </Story>
</Canvas>

### Warning

<Canvas withSource="open">
  <Story
    name="Warning"
    argTypes={{ ...Default.argTypes }}
    args={{
      type: 'warning',
      title: 'Warning',
      description:
        'Warning Variant! This is a short but descriptive message that indicates a negative or dangerous action.',
    }}
  >
    {Default.bind({})}
  </Story>
</Canvas>

## Customizable Slots <a id="customizable-slots" />

### `#close`

> This `close` slot can be used for creating your own custom close buttons like this one example below ðŸ‘‡

<Canvas withSource="open">
  <Story
    name="Custom Close Button"
    argTypes={{ ...Default.argTypes }}
    args={{
      type: 'info',
      title: 'With a custom button',
      description:
        'This alert box looks very similar to our default examples but it contains a custom close button in it!',
    }}
  >
    {DefaultWithSlots.bind({})}
  </Story>
</Canvas>

### Tips <a id="tips" />

ðŸ’¡ Use the title prop to provide a clear and concise message to the user.

ðŸ’¡ Use the description prop to provide more detailed information or instructions, if necessary.

ðŸ’¡ Choose the appropriate variant (success, error, warning, or info) for the Alert based on the content it is displaying.

ðŸ’¡ Keep the Alert small and concise. If you need to provide a lot of information, consider using a separate component or modal instead of trying to cram everything into the Alert.

ðŸ’¡ If the Alert is closable, make sure to provide clear instructions on how to dismiss it (e.g. "Click the 'X' to close this Alert").
