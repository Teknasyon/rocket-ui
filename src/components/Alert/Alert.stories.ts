import type { Meta, StoryObj } from '@storybook/vue3';
import Alert from './RAlert.vue';
import Icon from '../Icon/RIcon.vue';

const DefaultArgs = {
  type: 'success',
  title: 'Alert Title',
  description:
    'A type of alert which appears in a layer above other content, visually similar to a mobile or desktop push notification.',
  size: 'medium',
  closable: true,
  closeText: 'Close',
  block: false,
};

const DefaultArgTypes = {
  type: {
    type: 'select',
    options: ['success', 'info', 'warning', 'error'],
  },
  title: {
    type: 'text',
  },
  description: {
    type: 'text',
  },
  size: {
    type: 'select',
    options: ['small', 'medium', 'large'],
  },
  closable: {
    type: 'boolean',
  },
  block: {
    type: 'boolean',
  },
  closeText: {
    type: 'text',
  },
  onClose: {
    action: 'close',
  },
};

const AlertStory = {
  component: Alert,
  setup: (args: typeof Alert) => {
    return { args };
  },
  template: `<Alert v-bind="args" @close="click"/>`,
  args: DefaultArgs,
  argTypes: DefaultArgTypes,
} as Meta<typeof Alert>;

export default AlertStory;

type Story = StoryObj<typeof AlertStory>;

export const Overview: Story = {
  args: {
    type: 'success',
    title: 'Alert Title',
    description:
      'A type of alert which appears in a layer above other content, visually similar to a mobile or desktop push notification.',
  },
};

export const Success: Story = {
  args: {
    type: 'success',
    title: 'Success',
    description:
      'Success Variant! This is a short but descriptive message that indicates a successful action.',
  },
};

export const Error: Story = {
  args: {
    type: 'error',
    title: 'Error',
    description:
      'Error Variant! This is a short but descriptive message that indicates a negative action.',
  },
};

export const Information: Story = {
  args: {
    type: 'info',
    title: 'Info',
    description:
      'Info Variant! This is a short but descriptive message that indicates a neutral informative action.',
  },
};

export const Warning: Story = {
  args: {
    type: 'warning',
    title: 'Warning',
    description:
      'Warning Variant! This is a short but descriptive message that indicates a negative or dangerous action.',
  },
};

export const CustomCloseButton = {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  render: (args: any) => ({
    components: { Alert, Icon },
    setup: () => {
      return { args };
    },
    template: `
    <Alert
      v-bind="args"
    >
      <template #custom-close>
        <Icon name="mdiClose" />
      </template>
    </Alert>
    `,
  }),
  args: {
    type: 'info',
    title: 'With a custom button',
    description:
      'This alert box looks very similar to our default examples but it contains a custom close button in it!',
  },
};
