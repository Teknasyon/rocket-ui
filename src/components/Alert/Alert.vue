<script setup lang="ts">
import './alert.css';
import { computed } from 'vue';
import Icon from '../Icon/Icon.vue';

export interface Props {
  type: 'success' | 'error' | 'warning' | 'info';
  title?: string;
  description?: string;
  size?: 'small' | 'medium' | 'large';
  closable?: boolean;
}

const emit = defineEmits(['close']);
const props = withDefaults(defineProps<Props>(), {
  type: 'info',
  title: '',
  description: '',
  size: 'medium',
  closable: true,
});
const classes = computed(() => {
  return {
    alert: true,
    [`alert--${props.type}`]: true,
    [`alert--${props.size}`]: true,
  };
});
const icon = computed(() => {
  return {
    success: 'check_circle',
    error: 'error',
    warning: 'warning',
    info: 'info',
  }[props.type];
});
const iconSize = computed(() => {
  return {
    small: 12,
    medium: 16,
    large: 20,
  }[props.size];
});
const close = () => {
  emit('close');
};
</script>
<template>
  <div :class="classes">
    <div class="icon">
      <Icon :name="icon" :size="iconSize" />
    </div>
    <div class="texts">
      <p class="texts__title">{{ props.title }}</p>
      <p class="texts__description">{{ props.description }}</p>
    </div>
    <div v-if="props.closable" class="close">
      <Icon
        v-if="!$slots['close']"
        class="close__button"
        name="close"
        size="16"
        @click="close"
      />
      <slot name="close" v-else />
    </div>
  </div>
</template>
