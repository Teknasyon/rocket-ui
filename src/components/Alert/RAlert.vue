<script setup lang="ts">
import './alert.css'
import { computed } from 'vue'
import Icon from '../Icon/RIcon.vue'

export interface Props {
  /**
   * Type of the Alert
   * @type 'success' | 'error' | 'warning' | 'info'
   * @default 'info'
   * @example
   * <Alert type="success" />
   */
  type: 'success' | 'error' | 'warning' | 'info'

  /**
   * Title of the Alert
   * @type string
   * @default ''
   * @example
   * <Alert title="Title" />
   */
  title: string

  /**
   * Variant of the Alert
   * @type 'solid' | 'outline' | 'ghost'
   * @default 'ghost'
   * @example
   * <Alert variant="ghost" />
   */
  variant?: 'solid' | 'outline' | 'ghost'

  /**
   * Description of the Alert
   * @type string
   * @default ''
   * @example
   * <Alert description="Description" />
   */
  description?: string

  /**
   * Allow to close the Alert
   * @type boolean
   * @default true
   * @example
   * <Alert closable />
   */
  closable?: boolean

  /**
   * Block appearance of the Alert
   * @type boolean
   * @default false
   * @example
   * <Alert block />
   */
  block?: boolean
}

const props = withDefaults(defineProps<Props>(), {
  type: 'info',
  title: '',
  variant: 'ghost',
  description: '',
  closable: true,
  block: false,
})
const emit = defineEmits(['close'])
const classes = computed(() => {
  return {
    ' r-alert': true,
    [`r-alert--${props.variant}`]: true,
    'r-alert--block': props.block,
    [`r-alert--${props.variant}--${props.type}`]: true,
  }
})

const icon = computed(() => {
  return {
    success: 'mdiCheckCircle',
    error: 'mdiAlertCircle',
    warning: 'mdiAlert',
    info: 'mdiInformation',
  }[props.type]
})

function close() {
  emit('close')
}
</script>

<template>
  <div :class="classes">
    <div class="r-alert__icon">
      <slot name="alert-icon">
        <Icon :name="icon" :size="24" />
      </slot>
    </div>
    <div class="r-alert__content">
      <slot name="content">
        <p class="r-alert__content__title">
          {{ props.title }}
        </p>
        <p v-if="props.description" class="r-alert__content__description">
          {{ props.description }}
        </p>
      </slot>
    </div>
    <div v-if="props.closable" class="r-alert__close">
      <slot name="custom-close">
        <Icon
          class="r-alert__close__button"
          name="mdiClose"
          :size="16"
          @click="close"
        />
      </slot>
    </div>
  </div>
</template>
