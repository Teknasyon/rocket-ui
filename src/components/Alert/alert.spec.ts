import { describe, expect, it } from 'vitest'
import { mount } from '@vue/test-utils'
import Alert from './RAlert.vue'

describe('Alert', () => {
  it('should render correctly', () => {
    const wrapper = mount(Alert, {
      props: {
        type: 'success',
        title: 'Alert Title',
        description: 'Alert Description',
        size: 'small',
        closable: true,
      },
    })

    expect(wrapper.classes()).toContain('alert')
    expect(wrapper.find('.texts__title').text()).toBe('Alert Title')
    expect(wrapper.find('.texts__description').text()).toBe(
      'Alert Description',
    )
    expect(wrapper.find('.alert').classes()).toContain('alert--small')
    expect(
      wrapper
        .find('.alert')
        .trigger('click')
        .then(() => {
          expect(wrapper.emitted('close')).toBeTruthy()
        }),
    )
  })
})
