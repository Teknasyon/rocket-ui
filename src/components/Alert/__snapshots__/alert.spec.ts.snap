// Vitest Snapshot v1

exports[`Alert > should render correctly 1`] = `
"<div class=\\"alert alert-success alert--small\\">
  <div class=\\"icon\\"><span class=\\"material-icons-round\\" style=\\"font-size: 24px; font-weight: regular; display: inline-flex; align-items: center; justify-content: center;\\">check_circle</span></div>
  <div class=\\"texts\\">
    <div class=\\"texts__title\\">Alert Title</div>
    <div class=\\"texts__description\\">Alert Description</div>
  </div>
  <div class=\\"close\\"><button type=\\"button\\" class=\\"button button--primary button--small close__button\\"><span class=\\"material-icons button__icon button__icon--left\\" style=\\"font-size: 24px; font-weight: regular; display: inline-flex; align-items: center; justify-content: center;\\"></span>Close2</button></div>
</div>"
`;
